db-engine/src/main/java/com/splicemachine/db/iapi/types/SQLTimestamp.java

```diff
@@ -169,12 +172,57 @@ public int estimateMemoryUsage()
         return BASE_MEMORY_USAGE;
     } // end of estimateMemoryUsage
 
-    private static String format(Timestamp timestamp, String timeStampFormat) {
+    /**
+     * this also checks if the format is a valid fixed-size timestamp format
+     * see also https://docs.oracle.com/javase/8/docs/api/java/time/format/DateTimeFormatter.html
+     * we support:
+     * yyyy, yy, uuuu, uu, eee, EEE, MM, mm, dd, HH, hh, ss, a
+     * an arbitrary number of S, and other characters " ,.-/:"
+     * @param format
+     * @throws IllegalArgumentException if format is not supported
+     * @return length of format
+     */
+    public static int getFormatLength(String format) throws IllegalArgumentException
+    {
+        byte[] b = format.getBytes(Bytes.UTF8_CHARSET);
+        int repeat = 0;
+        char last = (char)b[0];
+        int length = b.length;
+        for(int i=0; i<b.length+1; i++) {
+            if (i != b.length && (char) b[i] == last) {
+                repeat++;
+                continue;
+            }
+            int r = repeat;
+            char l = last;
+            repeat = 1;
+            if( i != b.length)
+                last = (char) b[i];
+            if ((l == 'y' || l == 'u') && (r == 4 || r == 2)) continue;
+            else if (r == 2 &&  "MmdHhs".indexOf(l) != -1 ) continue;
+            else if (r == 3 && (l == 'e' || l == 'E')) continue;
+            else if (l == 'S') continue;
+            else if (r == 1) {
+                if (l == 'a') {
+                    length++;
+                    continue;
+                }
+                if(" ,.-/:".indexOf(l) != -1) continue;
+            }
+            throw new IllegalArgumentException("not supported format \"" + format + "\": '" + l + "' can't be repeated " + r + " times");
+        }
```

Comment:
i agree, however customer doesn't see this message currently since we don't check this at `call SYSCS_UTIL.SYSCS_SET_GLOBAL_DATABASE_PROPERTY( 'splice.function.timestampFormat', 'yyyy-MM-dd-HH.mm.ss.SSSSSSSS');` but only later where we can't throw . i created a task/bug for this: DB-10968